TMessagesProj/src/main/java/org/telegram/messenger/UnifiedPushReceiver.java

```diff
@@ -0,0 +1,83 @@
+package org.telegram.messenger;
+
+import android.content.Context;
+import android.os.SystemClock;
+
+import androidx.annotation.NonNull;
+
+import org.telegram.tgnet.ConnectionsManager;
+import org.unifiedpush.android.connector.MessagingReceiver;
+
+import java.util.concurrent.CountDownLatch;
+
+public class UnifiedPushReceiver extends MessagingReceiver {
+    @Override
+    public void onNewEndpoint(Context context, String endpoint, String instance){
+        Utilities.globalQueue.postRunnable(() -> {
+            SharedConfig.pushStringGetTimeEnd = SystemClock.elapsedRealtime();
+
+            PushListenerController.sendRegistrationToServer(PushListenerController.PUSH_TYPE_SIMPLE, endpoint);
+        });
+    }
+
+    @Override
+    public void onMessage(Context context, byte[] message, String instance){
+        final long receiveTime = SystemClock.elapsedRealtime();
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        AndroidUtilities.runOnUIThread(() -> {
+            if (BuildVars.LOGS_ENABLED) {
+                FileLog.d("UP PRE INIT APP");
+            }
+            ApplicationLoader.postInitApplication();
+            if (BuildVars.LOGS_ENABLED) {
+                FileLog.d("UP POST INIT APP");
+            }
+            Utilities.stageQueue.postRunnable(() -> {
+                if (BuildVars.LOGS_ENABLED) {
+                    FileLog.d("UP START PROCESSING");
+                }
+                for (int a = 0; a < UserConfig.MAX_ACCOUNT_COUNT; a++) {
```

Comment:
You can use the `instance` (https://github.com/Telegram-FOSS-Team/Telegram-FOSS/pull/715/files#diff-214ddbae5110be032e120290f658254085a490a2a5c1ca08e01542f3e95a9febR1520) to link to one account, to avoid fetching for all of them, but that's just an optimization

Reply:
the point is that a notification can be for any user, since in registerDevice (https://core.telegram.org/method/account.registerDevice) you specify also other_uids that includes all the users (https://github.com/Telegram-FOSS-Team/Telegram-FOSS/blob/master/TMessagesProj/src/main/java/org/telegram/messenger/MessagesController.java#L12730) and so you can't just fetch the current user

Reply:
ok, makes sens. It is still possible to have an endpoint per user if you want